fdbserver/TagPartitionedLogSystem.actor.cpp

```diff
@@ -2996,11 +3021,33 @@ struct TagPartitionedLogSystem : ILogSystem, ReferenceCounted<TagPartitionedLogS
 
 		wait(waitForAll(initializationReplies) || oldRouterRecruitment);
 
-		for (int i = 0; i < initializationReplies.size(); i++) {
-			logSystem->tLogs[0]->logServers[i] = makeReference<AsyncVar<OptionalInterface<TLogInterface>>>(
-			    OptionalInterface<TLogInterface>(initializationReplies[i].get()));
-			logSystem->tLogs[0]->tLogLocalities[i] = recr.tLogs[i].locality;
+		if (SERVER_KNOBS->TLOG_NEW_INTERFACE) {
+			state std::unordered_map<UID, ptxn::TLogInterface_PassivelyPull > id2Interface;
+			state int i = 0;
+			for (i = 0; i < reqs.size(); i++) {
+				ASSERT(reqs[i].isPrimary);
+				// wait for interfaces being built from TLogServer, then construct id -> interface mapping
+				// cannot use more standard `getReplyUnlessFailedFor`, because it is waiting on `reply` field, here we have ptxn.reply
+				state ptxn::TLogInterface_PassivelyPull serverNew = wait(reqs[i].ptxnReply.getFuture());
```

Comment:
Instead of waiting for reply here, we need to replace initializationReplies above with initializationPtxnReplies.

Reply:
* can you explain a bit more, why?
* I am not sure how it works from caller side -- i.e. which method is called when old_tlog or new_tlog is recruited